control_panel_api/filters.py

```diff
@@ -43,11 +43,27 @@ class S3BucketFilter(SuperusersOnlyFilter):
     pass
 
 
-class UserFilter(SuperusersOnlyFilter):
+class SelfOrSuperuserFilter(DjangoFilterBackend):
```

Comment:
As discussed offline we can probably have this as `UserFilter` as this logic is very specific of the user filtering.